### ISSUE TITLE
Crash when trying to configure [ACPCore configureWithAppId:] 

### ISSUE THREAD

**Issue by rsterst** [Adobe-Marketing-Cloud/acp-sdks#682]
As the title says it, we experience crashes for some users during ACPCore configuration. 

```
0   **********                     	0x000000010605ac1c std::__1::__tree_node_base<void*>*& std::__1::__tree<std::__1::__value_type<std::__1::__thread_id, unsigned long>, std::__1::__map_value_compare<std::__1::__thread_id, std::__1::__value_type<std::_... + 4
1   **********                     	0x000000010605abbc std::__1::pair<std::__1::__tree_iterator<std::__1::__value_type<std::__1::__thread_id, unsigned long>, std::__1::__tree_node<std::__1::__value_type<std::__1::__thread_id, unsigned long>, void*>*, l... + 40
2   **********                     	0x000000010604f658 AdobeMarketingMobile::StringUtils::GetCurrentDepth() + 80
3   **********                     	0x0000000105fc9040 void AdobeMarketingMobile::StringUtils::ToString<char [10]>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, char const (&) [10], AdobeMarketingMobile::ToStringOptions const&) + 32
4   **********                     	0x0000000105fc8f48 std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > AdobeMarketingMobile::StringUtils::ToString<char [10]>(char const (&) [10], AdobeMarketingMobile::ToStringOptio... + 184
5   **********                     	0x0000000105fc8d2c std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > AdobeMarketingMobile::StringUtils::ToString<char [10]>(char const (&) [10]) + 88
6   **********                     	0x0000000105fc8794 AdobeMarketingMobile::(anonymous namespace)::Statics::Get() + 268
7   **********                     	0x0000000105fc8164 AdobeMarketingMobile::MobileMarketingCoreBridge::GetInstance2() + 48
8   **********                     	0x0000000105fcd424 GetCore() + 44
9   **********                     	0x0000000105fcd308 +[ACPCore configureWithAppId:] + 40
```

**Comment by cdhoffmann** [comment]
@rsterst Hello. Could you please share the full crash report a long with the following information:

1. Versions for ACPCore and all other Adobe Extensions being used
2. Are you properly implementing the configureWithAppID call in conjunction with the registration and start calls as shown [here.](https://developer-stage.adobe.com/client-sdks/previous-versions/documentation/mobile-core/api-reference/#registerextensions)
3. How often is this happening? Is it reproducible locally?

**Comment by praveek** [comment]
@rsterst We have identified a potential fix for this crash and are planning to release an update next week. I will update this issue after the release.

**Comment by praveek** [comment]
@rsterst We released [ACPCore v2.9.7](https://github.com/Adobe-Marketing-Cloud/acp-sdks/releases/tag/v2.9.7-ACPCore) which addresses this issue.
